monai/engines/workflow.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+import torch
+from ignite.engine import Engine, State, Events
+from .utils import default_prepare_batch
+
+
+class Workflow(ABC):
```

Comment:
I'd suggest this class should inherit `ignite.engine.Engine` instead of having an internally created instance of it. This would simplify code so that `self.engine` wouldn't be needed everywhere, and any existing code expecting `Engine` instances will function just by passing in a `Workflow` instance without having to type check.

Reply:
updated to inherit Engine in the latest patch.
Thanks.